msu/systems/mod_settings/mod_settings_system.nut

```diff
@@ -87,8 +87,12 @@
 			}
 		}
 		*/
+		local persistentData = ::MSU.Mod.PersistentData.hasFile("ModSettings") ? ::MSU.Mod.PersistentData.readFile("ModSettings") : {};
```

Comment:
Instead of doing this and the below logic, I think we should map get all the updated settings, then filter based on getPersistence() and then feed this to `exportSettingsToPersistentData`